widget/playground/src/utils/common.ts

```diff
@@ -17,3 +25,103 @@ export function shallowEqual<T>(
 
   return true;
 }
+
+export const tokenToString = (token: Asset) =>
+  `${token.symbol}-${token.blockchain}-${token.address ?? ''}`;
+
+export function removeDuplicates<T>(arr: T[], key?: keyof T): T[] {
+  if (!key) {
+    return Array.from(new Set(arr));
+  }
+
+  const seen = new Set<T[keyof T]>();
+  return arr.filter((item) => {
+    const identifier = item[key];
+    if (!seen.has(identifier)) {
+      seen.add(identifier);
+      return true;
+    }
+    return false;
+  });
+}
+
+export function tokensAreEqual(tokenA?: Asset, tokenB?: Asset) {
+  return (
+    tokenA?.blockchain === tokenB?.blockchain &&
+    tokenA?.symbol === tokenB?.symbol &&
+    tokenA?.address === tokenB?.address
+  );
+}
+
+export const containsText = (text: string, searchText: string) =>
+  text.toLowerCase().indexOf(searchText.toLowerCase()) > NOT_FOUND;
+
+export const filterTokens = (list: Token[], searchedFor: string) =>
+  list.filter(
+    (token) =>
+      containsText(token.symbol, searchedFor) ||
+      containsText(token.address || '', searchedFor) ||
+      containsText(token.name || '', searchedFor)
+  );
+
+export const excludedWallets = [
+  WalletTypes.STATION,
+  WalletTypes.LEAP,
+  WalletTypes.SAFE,
+  WalletTypes.MY_TON_WALLET,
+  WalletTypes.WALLET_CONNECT_2,
+  WalletTypes.BINANCE_CHAIN,
+];
+
+export const onChangeMultiSelects = (
+  value: string,
+  values: any[] | undefined,
+  list: any[],
+  findIndex: (item: string) => boolean
+): string[] | undefined => {
+  if (value === 'empty') {
+    return [];
+  } else if (value === 'all') {
+    return undefined;
+  }
+  if (!values) {
+    values = [...list];
+    const index = list.findIndex(findIndex);
+    values.splice(index, 1);
+    return values;
+  }
+  values = [...values];
+  const index = values.findIndex(findIndex);
+  if (index !== NOT_FOUND) {
+    values.splice(index, 1);
+  } else {
+    values.push(value);
+  }
+  if (values.length === list.length) {
+    return undefined;
+  }
+  return values;
```

Comment:
`  return targetArray.length === list.length ? undefined : targetArray;
`

Reply:
As I mentioned above, this function will be removed.